aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SyncProcessor.java

```diff
@@ -126,16 +126,39 @@ Completable hydrate() {
             TopologicalOrdering.forRegisteredModels(schemaRegistry, modelProvider);
         Collections.sort(modelSchemas, ordering::compare);
         ArrayList<String> toBeSyncedModelArray = new ArrayList<>();
+        boolean syncInParallel = true;
```

Comment:
[nit] I think there may be slight differences in the meaning of "concurrent" and "parallel", and so it may be better to keep the terminology the same, by naming it `syncConcurrently` or `canSyncConcurrently` since the configuration name is "syncMaxConcurrentModels". Later the code will read

"if canSyncConcurrently && syncMaxConcurrentModels > 1"